src/language/dafnyInstallation.ts

```diff
@@ -123,6 +129,32 @@ export class DafnyInstaller {
     }
   }
 
+  private async installFromSource() {
+    const { execSync } = require('child_process');
+    const execLog = (command: string) => {
+      this.writeStatus(`Executing: ${command}\n`);
+      execSync(command);
+    };
+    const installationPath = this.getCustomInstallationPath(os.arch());
+    await mkdirAsync(installationPath.fsPath, { recursive: true });
+    this.writeStatus(`Installing Dafny from source in ${installationPath.fsPath}.\n`);
+    const process = require('process');
+    process.chdir(installationPath.fsPath);
+    execLog('brew install dotnet-sdk');
+    execLog('git clone --recurse-submodules https://github.com/dafny-lang/dafny.git');
+    process.chdir(Utils.joinPath(installationPath, 'dafny').fsPath);
+    execLog('make exe');
+    const Binaries = Utils.joinPath(installationPath, 'dafny', 'Binaries').fsPath;
+    process.chdir(Binaries);
+    execLog('brew install wget');
+    execLog('wget https://github.com/Z3Prover/z3/releases/download/Z3-4.8.5/z3-4.8.5-x64-osx-10.14.2.zip');
+    execLog('unzip z3-4.8.5-x64-osx-10.14.2.zip');
+    execLog('mv z3-4.8.5-x64-osx-10.14.2 z3');
+    process.chdir(this.getInstallationPath().fsPath);
+    execLog(`cp -R ${Binaries}/* ./dafny/`);
```

Comment:
You may want to check the return codes of the commands to not proceed when any step fails. On top of that you could provide some meaningful help message.

Reply:
I don't think this is necessary, as `exec` will throw an exception if the return code is not zero.
https://stackoverflow.com/questions/32874316/node-js-accessing-the-exit-code-and-stderr-of-a-system-command